master/reference/node/configuration.md

```diff
@@ -51,6 +51,22 @@ In addition to the above, `{{site.nodecontainer}}` also supports [the standard F
 >
 {: .alert .alert-info}
 
+### Node name determination
+
+The `{{site.nodecontainer}}` must know the name of the node on which it is running. The node name is used to
+retrieve the [Node resource](../calicoctl/resources/node) configured for this node if it exists, or to create a new node resource representing the node if it does not. It is
+also used to associate the node with per-node [BGP configuration](../calicoctl/resources/bgpconfig), [felix configuration](../calicoctl/resources/felixconfig), and endpoints.
+
+When launched, the `{{side.nodecontainer}}` container uses the following methods in order to determine
+what name to use:
+
+1. The value specified in the `NODENAME` environment variable, if set.
+1. The value specified at `/var/lib/calico/nodename` if it exists.
+1. The value specified in the `HOSTNAME` environement variable, if set.
+1. The hostname as returned by the OS, converted to lowercase.
+
```

Comment:
recommend: "OS" --> "operating system"

I would also add a "for example" here to show exactly how the order of precedence works. "For example, if the operating system returns a hostname of `amazingserver1` and the `NODENAME` environment variable contains a value of `moreamazingserver5`, {{site.nodecontainer}}` sets the node name to `moreamazingserver5`."